src/Model/ModelManagerInterface.php

```diff
@@ -338,9 +338,11 @@ public function getExportFields($class);
     public function getPaginationParameters(DatagridInterface $datagrid, $page);
 
     /**
-     * @param string $class
+     * @param string                 $class
+     * @param array<int, int|string> $idx
      *
-     * @phpstan-param class-string $class
+     * @phpstan-template T of object
```

Comment:
I don't think adding this Template is useful since it's used only once in this Phpdoc.

Reply:
I just copied the same approach used by other methods:
https://github.com/sonata-project/SonataAdminBundle/blob/44ee77dcd24edf8e2a7a04fa5ffaa664bc8fe54b/src/Model/ModelManagerInterface.php#L84-L94

https://github.com/sonata-project/SonataAdminBundle/blob/44ee77dcd24edf8e2a7a04fa5ffaa664bc8fe54b/src/Model/ModelManagerInterface.php#L72-L82

https://github.com/sonata-project/SonataAdminBundle/blob/44ee77dcd24edf8e2a7a04fa5ffaa664bc8fe54b/src/Model/ModelManagerInterface.php#L60-L70

How should this docblock look based on your proposal?

Reply:
For the other docBlock, there is a `phpstan-return`. Here the method returns nothing.

When you use
```
* @phpstan-template T of object 
* @phpstan-param class-string<T> $class 
* @phpstan-return T[] 
```
You're saying that you want a `class-string`, and if phpstan know that you're passing `Foo::class`, it'll know that the result is a `Foo[]`.

In the code you modified, there is only
```
* @phpstan-template T of object 
* @phpstan-param class-string<T> $class
```
You're saying that you want a `class-string`, and phpstan does not use the extra Template information.

If the docblock was
```
* @phpstan-template T of object 
* @phpstan-param class-string<T> $class
* @phpstan-param class-string<T> $class2
```
It would have a sens, it would have meant that the two class should be the same class-string.

In a general way, if you use the `T` value only once, it's useless to add it.

So I would just keep 
```
* @phpstan-param class-string $class
```

Reply:
Thank you for the detailed explanation.